component/src/main/java/org/wso2/extension/siddhi/gpl/execution/pmml/PmmlModelProcessor.java

```diff
@@ -186,66 +140,70 @@ protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processo
 
-        List<Attribute> outputAttributes = new ArrayList<>();
-        int numOfOutputFields = evaluator.getOutputFields().size();
-        for (FieldName field : outputFields) {
-            String dataType;
-            if (numOfOutputFields == 0) {
-                dataType = evaluator.getDataField(field).getDataType().toString();
-            } else {
-                // If dataType attribute is missing, consider dataType as string(temporary fix).
-                if (evaluator.getOutputField(field).getDataType() == null) {
-                    logger.info("Attribute dataType missing for OutputField. Using String as dataType");
-                    dataType = "string";
-                } else {
-                    dataType = evaluator.getOutputField(field).getDataType().toString();
-                }
+        for (Map.Entry<InputField, int[]> entry : attributeIndexMap.entrySet()) {
+            InputField inputField = entry.getKey();
+            int[] attributeIndexArray = entry.getValue();
+            Object dataValue = null;
+            switch (attributeIndexArray[2]) {
+                case 0:
+                    dataValue = event.getBeforeWindowData()[attributeIndexArray[3]];
+                    break;
+                case 2:
+                    dataValue = event.getOutputData()[attributeIndexArray[3]];
+                    break;
+                default:
+                    break;
             }
-            Attribute.Type type = null;
-            if (dataType.equalsIgnoreCase("double")) {
-                type = Attribute.Type.DOUBLE;
-            } else if (dataType.equalsIgnoreCase("float")) {
-                type = Attribute.Type.FLOAT;
-            } else if (dataType.equalsIgnoreCase("integer")) {
-                type = Attribute.Type.INT;
-            } else if (dataType.equalsIgnoreCase("long")) {
-                type = Attribute.Type.LONG;
-            } else if (dataType.equalsIgnoreCase("boolean")) {
-                type = Attribute.Type.BOOL;
-            } else if (dataType.equalsIgnoreCase("string")) {
-                type = Attribute.Type.STRING;
+            inData.put(inputField.getName(), inputField.prepare(String.valueOf(dataValue)));
+        }
+
+        if (!inData.isEmpty()) {
+            try {
+                Map<FieldName, ?> result = evaluator.evaluate(inData);
+                Object[] output = new Object[outputFields.size()];
+                int i = 0;
+                for (FieldName fieldName : outputFields.keySet()) {
+                    if (result.containsKey(fieldName)) {
+                        Object value = result.get(fieldName);
+                        output[i] = EvaluatorUtil.decode(value);
+                        i++;
+                    }
+                }
+                complexEventPopulater.populateComplexEvent(event, output);
+                nextProcessor.process(streamEventChunk);
+            } catch (Exception e) {
```

Comment:
Why we catch exception here ? and can we give more context to the exception message ?